solr/core/src/java/org/apache/solr/util/circuitbreaker/CircuitBreakerRegistry.java

```diff
@@ -62,9 +81,11 @@ public void deregisterAll() {
    */
   public List<CircuitBreaker> checkTripped(SolrRequestType requestType) {
     List<CircuitBreaker> triggeredCircuitBreakers = null;
+    final List<CircuitBreaker> breakersOfType = new ArrayList<>();
 
-    for (CircuitBreaker circuitBreaker :
-        circuitBreakerMap.getOrDefault(requestType, Collections.emptyList())) {
+    breakersOfType.addAll(circuitBreakerMap.getOrDefault(requestType, Collections.emptyList()));
+    breakersOfType.addAll(globalCircuitBreakerMap.getOrDefault(requestType, Collections.emptyList()));
+    for (CircuitBreaker circuitBreaker : breakersOfType) {
```

Comment:
I'm sure it would be possible to stream the two maps in a more efficient streaming fashion to avoid extra list, but this is just a POC :)